lib/stringify.js

```diff
@@ -80,7 +80,7 @@ var stringify = function stringify(
     var tmpSc = sideChannel;
     var step = 0;
     var findFlag = false;
-    while ((tmpSc = tmpSc.get(sentinel)) !== undefined && !findFlag) {
+    while (typeof (tmpSc = tmpSc.get(sentinel)) !== 'undefined' && !findFlag) {
```

Comment:
altho this one is actually a bit trickier because of the assignment in the loop, so let's do this:
```suggestion
    while ((tmpSc = tmpSc.get(sentinel)) !== void undefined && !findFlag) {
```